sky/cli.py

```diff
@@ -3063,16 +3063,35 @@ def _output():
                 return
 
         # Show detailed accelerator information
+
+        name, quantity = accelerator_str.split(':')[0], None
+        has_quantity = accelerator_str.count(':') > 0
+        if has_quantity:
+            quantity = int(accelerator_str.split(':')[1])
         result = service_catalog.list_accelerators(gpus_only=True,
-                                                   name_filter=gpu_name,
+                                                   name_filter=name,
+                                                   quantity=quantity,
                                                    region_filter=region,
                                                    clouds=cloud)
         if len(result) == 0:
-            yield f'Resources \'{gpu_name}\' not found. '
+            yield f'Resources \'{name}\' not found. '
             yield 'Try \'sky show-gpus --all\' '
             yield 'to show available accelerators.'
             return
 
+        num_accelerators_found = 0
+        num_unique_accelerators = len([1 for i in result if len(result[i]) > 0])
+        for i in result:
+            num_accelerators_found += len(result[i])
+        if num_accelerators_found == 0:
```

Comment:
We shouldn't need these checks - result should be empty if no accelerators match the quantity_filter. We should modify `list_accelerators` have this behavior.

Reply:
Removed all this unnecessary filtering